black-jack-redo.rb

```diff
@@ -25,88 +25,205 @@ def print_with_stars(string)
 	puts string
 	puts '*' * 60
 	puts '*' * 60
+	puts '--------'
 end
 
-def viewhand(array, string)
-	count3 = 0
-	count4 = 0
-	while count3 < array.length
-		if array[count3][:dealt] == true
-			string += array[count3][:card] + ' of ' + array[count3][:suit] + "\n"
-			count4 += array[count3][:value]
-		else
-		end
-		count3 += 1
+def get_suit(string)
+	card_suit = ''
+	if string[0] == 'S'
+		card_suit = "Spades\n"
+	elsif string[0] == 'H'
+		card_suit = "Hearts\n"
+	elsif string[0] == 'D'
+		card_suit = "Diamonds\n"
+	elsif string[0] == 'C'
+		card_suit = "Clubs\n"
+	else 
+		puts 'Something went wrong!'
 	end
-	return count4, string
+
+	return card_suit
+end
+
+def get_card(card)
+	user_card = ''
+	if (card[1] == 'A') || (card[1] == 'a')
+		user_card = 'Ace of ' + get_suit(card)
+	elsif card[1] == 'J'
+		user_card = 'Jack of ' + get_suit(card)
+	elsif card[1] == 'Q'
+		user_card = 'Queen of ' + get_suit(card)
+	elsif card[1] == 'K'
+		user_card =  'King of ' + get_suit(card)
+	elsif (card[1].to_i >= 2) || (card[1..-1].to_i <= 10)
+		user_card = "#{card[1..-1]} of " + get_suit(card)
+	else
+		puts 'Something went wrong when retrieving the card!'
+		return
+	end
+	return user_card
 end
 
-def handcount(array)
-	count3 = 0
-	count4 = 0
-	while count3 < array.length
-		if array[count3][:dealt] == true
-			count4 += array[count3][:value]
+def viewhand(array, is_user)
+	count = 0
+	hand_value = 0
+	if is_user == 'true'
+		user_hand = "You have #{array.length} cards in your hand! " + "\n" + "Your hand includes the following:" + "\n"
+	else
+		user_hand = "The dealer has:" + "\n"
+	end
+
+	while count < array.length
+		string = array[count]
+		if string[1] == 'A'
+			user_hand = user_hand + 'Ace of ' + get_suit(array[count])
+			hand_value += 11
+			count += 1
+		elsif string[1] == 'a'
+			user_hand = user_hand + 'Ace of ' + get_suit(array[count])
```

Comment:
You have a lot of repetition in these else statements. Condense `J`, `Q,`, and `K` into one conditional.